app/controllers/applications_controller.rb

```diff
@@ -0,0 +1,47 @@
+class ApplicationsController < ApplicationController
+  def show
+    @application = Application.find(params[:id])
+
+    if params[:search].present?
+      @pets = Pet.where('name ILIKE ?', "%#{params[:search]}%")
+    end
+  end
+
+  def new
+  end
+
+  def submit
+    @application = Application.find(params[:id])
+    @application.update({"status" => "Pending"})
+    redirect_to "/applications/#{@application.id}"
+  end
+
+  def create
+    @application = Application.new(application_params)
+    @application.status = 'In Progress'
+    if @application.save
+      redirect_to "/applications/#{@application.id}"
+    else
+      flash[:error] = "All fields must be filled in."
```

Comment:
To answer your question about whether `"All fields must be filled in."` should go in the view since it seems to be view logic. The quick answer to that is no, we are storing that message into the `flash` hash. We then have code in our view that will look in that `flash` hash to see if there is an `:error` key and display the error, that is the view logic... aka using logic to determine if something should show up on the page.

To answer the second part of that question, similar to how the `params` hash is part of our controller, so is the `flash` hash, therefore you do not need to extract it to the model. It's available for you to use to display messages to the User, there is no data logic involved.